test/test_scandir.py

```diff
@@ -250,6 +250,23 @@ def test_unicode(self):
         self.assertEqual(entry.name, 'file1.txt')
         self.assertEqual(entry.path, os.path.join(path, 'file1.txt'))
 
+    def test_os_walk_unicode_handling(self):
+        if sys.platform == 'win32':
+            return self.skipTest("Not windows")
+        subdir = os.path.join(TEST_PATH, 'walk').encode('utf-8')
```

Comment:
os.fsencode() isn't available on py2.x. I'll use sys.getfilesystemencoding. Encoding TEST_PATH does not fail on py33.